builder/test/copy.cc

```diff
@@ -384,6 +384,54 @@ TEST_P(upsample_y, copy) {
   ASSERT_EQ(eval_ctx.copy_elements, W * H);
 }
 
+
+class downsample_y : public testing::TestWithParam<int> {};
+
+INSTANTIATE_TEST_SUITE_P(split, downsample_y, testing::Range(0, 5));
+
+TEST_P(downsample_y, copy) {
+  int split = GetParam();
+
+  // Make the pipeline
+  node_context ctx;
+
+  auto in = buffer_expr::make(ctx, "in", 2, sizeof(int));
+  auto out = buffer_expr::make(ctx, "out", 2, sizeof(int));
+
+  var x(ctx, "x");
+  var y(ctx, "y");
+
+  func downsample = func::make_copy({in, {point(x), point(y * 2)}}, {out, {x, y}});
+
+  if (split > 0) {
+    downsample.loops({{y, split}});
+  }
+
+  pipeline p = build_pipeline(ctx, {in}, {out});
+
+  // Run the pipeline.
+  const int H = 20;
+  const int W = 10;
+  buffer<int, 2> in_buf({W, H * 2});
+  init_random(in_buf);
+
+  buffer<int, 2> out_buf({W, H});
+  out_buf.allocate();
+  const raw_buffer* inputs[] = {&in_buf};
+  const raw_buffer* outputs[] = {&out_buf};
+  test_context eval_ctx;
+  p.evaluate(inputs, outputs, eval_ctx);
+
+  for (int y = 0; y < H; ++y) {
+    for (int x = 0; x < W; ++x) {
+      ASSERT_EQ(out_buf(x, y), in_buf(x, y * 2));
+    }
+  }
+
+  ASSERT_EQ(eval_ctx.copy_calls, split == 0 ? 1 : (H + split / 2) / split);
```

Comment:
Shouldn't this be a ceiling, not a round? Also, we have a helper for this: `ceil_div(H, split)`

Reply:
This should be ceiling.